sway/desktop/output.c

```diff
@@ -745,12 +745,18 @@ static void damage_child_views_iterator(struct sway_container *con,
 
 void output_damage_whole_container(struct sway_output *output,
 		struct sway_container *con) {
+	int shadow_sigma = con->shadow_enabled ? config->shadow_blur_sigma : 0;
+	int blur_size = con->blur_enabled ?
+			pow(2, config->blur_params.num_passes) * config->blur_params.radius : 0;
+	// +1 as a margin of error
+	int effect_size = MAX(shadow_sigma, blur_size) + 1;
+
```

Comment:
Move into `output_repaint_timer_handler`?

Reply:
What's your thinking with that change?

Reply:
It would make it easier to separate the two different damages needed (expanded once and double expanded damages). The current way you're doing would be a lot cleaner but I'm not sure how we're gonna separate the two? Any ideas?

Reply:
Why do we double expand?

Reply:
> Why do we double expand?

The first expand is due to how the blur uses the surrounding pixels to blur, the second is to remove the vignette created at the edges. When we render to the wlr buffer we only render the inner single expanded damage. So we just push the vignette outwards